src/server/game/Handlers/CharacterHandler.cpp

```diff
@@ -2274,7 +2274,8 @@ void WorldSession::HandleCharRaceOrFactionChangeCallback(std::shared_ptr<WorldPa
         trans->Append(stmt);
 
         // Race specific languages
-        if (factionChangeInfo->RaceID != RACE_ORC && factionChangeInfo->RaceID != RACE_HUMAN && factionChangeInfo->RaceID != RACE_MAGHAR_ORC)
+        static const std::set<uint8> racesWithoutSpecificLanguages = { RACE_ORC, RACE_HUMAN, RACE_MAGHAR_ORC, RACE_KUL_TIRAN };
```

Comment:
use std::array, dont make it static and use std::find on it

Reply:
Can do, is there a particular reason to not use static though?

Reply:
Not a strong reason - using static local variables might introduce thread-safe initialization code for that variable (wasn't happening with std::array but was with std::set, and that one was also allocating memory with `new`). Since we don't need actually need this to be a shared "global" variable, a non-static one works just as well